.ci/check-performance-regression.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+set -e
+
+# max difference tolerance in %
+THRESHOLD_PERCENTAGE=10
+# baseline of execution time in seconds
+BASELINE_SECONDS=415.72
+
+# sample project path
+SAMPLE_PROJECT="./.ci-temp/jdk17"
+CONFIG_FILE="./config/benchmark-config.xml"
+
+# execute a command and time it
+# $TEST_COMMAND: command being timed
+time_command() {
+  # execute the command and time it
+  /usr/bin/time -o time.temp -q -f "%e" "$@" &>result.tmp
+
+  cat time.temp
+}
+
+# execute the benchmark a few times to calculate the average metrics
+# $JAR_PATH: path of the jar file being benchmarked
+execute_benchmark() {
+  local JAR_PATH=$1
+  if [ -z "$JAR_PATH" ]; then
+      echo "Missing JAR_PATH as an argument."
+      exit 1
+  fi
+
+  local TOTAL_SECONDS=0
+  local NUM_EXECUTIONS=3
+
+  [ ! -d "$SAMPLE_PROJECT" ] &&
+    echo "Directory $SAMPLE_PROJECT DOES NOT exist." | exit 1
+
+  # add suppressions to config file
+  sed -i '/  <!-- Filters -->/r ./config/projects-to-test/openjdk17-excluded.files' \
```

Comment:
Please extract this filename to a variable declared above with SAMPLE_PROJECT, etc., and do anything else to make it easy/obvious to update this script to run on any java version.

@Lmh-java how did we settle on 17, instead of 21?

Reply:
I am good with 17 for now, just curious about why we did this.

Reply:
I am still working on #14779 right now. If that's merged, we can directly use that new property to skip all exceptions. That will be faster than a long list of suppressions :).

Reply:
Done